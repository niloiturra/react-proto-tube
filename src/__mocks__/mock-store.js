import { createStore, applyMiddleware } from 'redux';
import { logger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';

import reducers from '../redux/modules';
import rootSaga from '../redux/sagas';

export function createTestStore() {
  const saga = createSagaMiddleware();
  const middlewares = [saga, logger];
  const store = createStore(reducers, applyMiddleware(...middlewares));
  saga.run(rootSaga);

  return store;
}
