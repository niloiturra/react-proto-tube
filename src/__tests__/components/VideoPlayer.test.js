import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import VideoPlayer from '../../components/VideoPlayer';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
// eslint-disable-next-line jest/no-mocks-import
import { createTestStore } from '../../__mocks__/mock-store';

Enzyme.configure({ adapter: new Adapter() });

let store;
describe('testing VideoPlayer component render', () => {
  beforeEach(() => {
    store = createTestStore();
  });

  it('should render VideoPlayer component correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <VideoPlayer />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
