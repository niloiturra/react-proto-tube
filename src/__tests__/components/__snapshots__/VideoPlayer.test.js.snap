// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing VideoPlayer component render should render VideoPlayer component correctly 1`] = `ShallowWrapper {}`;
